geostreams/client.py

```diff
@@ -1,6 +1,70 @@
 import os
+from collections import Mapping, Iterable
 import redis
-import numpy
+import numpy as np
+from zict import Func
+from zict.common import ZictBase
+
+
+def _nested_decode(tree):
+    # TODO
+    stack = [tree]
+    output = [[]]
+
+    while len(stack) > 0:
+        item = stack.pop()
+        out = output.pop()
+        if isinstance(item, Mapping):
+            out.append()
+
+
+class Redis(ZictBase):
+
+    def __init__(self, r: redis.StrictRedis = None):
+        "docstring"
+        if r is None:
+            r = redis_connection()
+
+        self.redis_connection = r
+
+    def keys(self):
+        return self.redis_connection.keys('*')
+
+    __iter__ = keys
+
+    def __len__(self):
+        return self.redis_connection.dbsize()
+
+    def __getitem__(self, key):
+        key_type = self.redis_connection.type(key).decode("utf-8")
+        if key_type == 'string':
+            return self.redis_connection.get(key)
+        elif key_type == 'list':
+            return self.redis_connection.lrange(key, 0, -1)
+        elif key_type == 'hash':
+            return self.redis_connection.hgetall(key)
+
+    def __setitem__(self, key, value):
+
+        del self[key]
```

Comment:
Is it possible that this key will not exist before calling `setitem`? If so, use `_ = self.pop(key, None)` to avoid a `KeyError`.

Reply:
good catch.

Reply:
actually. this code is fine. __del__ calls redis.delete, which does not throw an error if the key does not exist.